### ISSUE TITLE
load_file not properly parsing scheme file

### ISSUE THREAD

**Issue by Habush** [opencog/atomspace#2847]
I'm using the `load_file` function in `<opencog/persist/file/fast_load.h>` to load sample scheme file containing atomese code and it gives incorrect output. I am using the following snippet to test the behavior.

```
#include <iostream>
#include <opencog/util/Logger.h>
#include <opencog/atoms/base/Handle.h>
#include <opencog/atomspace/AtomSpace.h>
#include <opencog/persist/file/fast_load.h>
#include <opencog/atoms/atom_types/atom_types.h>

using namespace std;
using namespace opencog;



int main(int argc, char *argv[]) 
{
	
	string path = "test.scm";
	AtomSpace as;
	load_file(path, as);
	
	cout << "Atomspace size: " << as.get_size() << endl;
	
	
	HandleSeq concepts;
	
	as.get_handles_by_type(concepts, CONCEPT_NODE);
	
	cout << "Num concept nodes: " << concepts.size() << endl;
}
```

Contents of `test.scm` file:

```
(Concept "f1" (stv 0.6 1))
(Concept "f2" (stv 0.5 1))


(Implication (stv 0.9 1)
   (Concept "f3" (stv 0.4 1))
   (Concept "f4" (stv 0.3 1)))

(Implication (stv 0.6 1)
    (And
        (Concept "f1" (stv 0.6 1))
        (Concept "f2" (stv 0.5 1)))
   (Concept "f5" (stv 0.2 1))
)

(Concept "f7" (stv 0.45 1))

(Implication (stv 0.8 1)
    (Concept "f1" (stv 0.6 1))
    (Concept "f11" (stv 0.95 1)))

(Implication (stv 0.5 1)
    (Concept "f9" (stv 0.68 1))
    (Concept "f8" (stv 0.02 1)))

(Implication (stv 0.4 1)    
    (And
        (Concept "f6" (stv 0.2 1))
        (Concept "f10" (stv 0.81 1)))
    (Concept "f12" (stv 0.01 1)))

(Implication (stv 0.25 1)
   (Concept "f13" (stv 0.75 1))
   (Concept "f3" (stv 0.4 1)))

```

The above code returns the following (erroneous) output:

```
Atomspace size: 4
Num concept nodes: 3
```

where it should have been

```
Atomspace size: 21
Num concept nodes: 13 
```

Note: this was working few weeks ago before I updated the atomspace code. The recent commits to `Sexpr` code must have caused this.

**Comment by linas** [comment]
Yes, I can reproduce this. 

Build example with `c++  -std=c++1z fail.cc -latomspace -L /usr/local/lib/opencog -latom_types -lload_scm`

**Comment by linas** [comment]
Fixed in #2848 -- the recent changes assumed that  the `(stv)` for links only occurs at the end, and never in the middle. I altered the code to look for them in the middle.

Closing because I'm pretty sure that fixes this.
